flytekit/core/task.py

```diff
@@ -378,6 +381,21 @@ def wrapper(fn: Callable[P, Any]) -> PythonFunctionTask[T]:
         return wrapper
 
 
+def _deduce_cache_version(fn: Callable[P, Any]) -> str:
```

Comment:
If the task uses an imported function, any updates to the imported function would not be captured by `inspect.getsource`:

```python
from utils import helper

@task(cache=True)
def my_task(x: int):
    x = my_helper(x)
```

I think this is okay as long as we documented the behavior and what users should do.

Reply:
Yup. I think it's okay too. Could we also update the docstring for cache_version [here](https://github.com/flyteorg/flytekit/blob/a366653e697cd3adf429a05adde2ac421fbcae99/flytekit/core/base_task.py#L116)?